app/src/ui/index.tsx

```diff
@@ -26,6 +27,18 @@ if (__DEV__) {
   installDevGlobals()
 }
 
+
+if (__DARWIN__) {
+  if (shellNeedsPatching(process)) {
+    const env = getEnvironmentFromShell()
+    if (env) {
+      const hack = process as any
+      hack._originalEnv = process.env
```

Comment:
I'm somewhat 👎  on porting this code over from the Atom PR, in case something starts to depend on this. Thoughts @desktop/core?